roles/lib_dyn/library/dyn_record.py

```diff
@@ -0,0 +1,271 @@
+                zone=module.params['zone']
+            )
+        else:
+            module.fail_json(msg="Unable to retrieve zone",
+                             error=str(error))
+
+
+    # To retrieve the node object we need to remove the zone name from the FQDN
+    dyn_node_name = module.params['record_fqdn'].replace('.' + module.params['zone'], '')
+
+    # Retrieve the zone object from dyn
+    dyn_zone = Zone(module.params['zone'])
+
+    # Retrieve the node object from dyn
+    dyn_node = dyn_zone.get_node(node=dyn_node_name)
+
+    # All states will need a list of the exiting records for the zone.
+    dyn_node_records = get_any_records(module, dyn_node)
+
+    if module.params['state'] == 'list':
+        module.exit_json(changed=False,
+                         records=get_record_values(
+                             dyn_node_records,
+                         ))
+
+    if module.params['state'] == 'present':
+
+        # First get a list of existing records for the node
+        values = get_record_values(dyn_node_records)
+        value_key = get_record_key(module.params['record_type'])
+
+        # Check to see if the record is already in place before doing anything.
+        if (dyn_node_records and
+                dyn_node_records[value_key][0].ttl == module.params['record_ttl'] and
+                module.params['record_value'] in values[value_key]):
+
+            module.exit_json(changed=False)
+
+
+        # Working on the assumption that there is only one record per
+        # node we will first delete the node if there are any records before
+        # creating the correct record
+        if dyn_node_records:
+            dyn_node.delete()
+
+        # Now lets create the correct node entry.
+        dyn_zone.add_record(dyn_node_name,
+                            module.params['record_type'],
+                            module.params['record_value'],
+                            module.params['record_ttl']
+                           )
+
+        # Now publish the zone since we've updated it.
+        dyn_zone.publish()
+        module.exit_json(changed=True,
+                         msg="Created node %s in zone %s" % (dyn_node_name, module.params['zone']))
+
+    if module.params['state'] == 'absent':
+        # If there are any records present we'll want to delete the node.
+        if dyn_node_records:
+            dyn_node.delete()
+            # Publish the zone since we've modified it.
+            dyn_zone.publish()
+            module.exit_json(changed=True,
+                             msg="Removed node %s from zone %s" % (dyn_node_name, module.params['zone']))
+        else:
+            module.exit_json(changed=False)
+
+# Ansible tends to need a wild card import so we'll use it here
+# pylint: disable=redefined-builtin, unused-wildcard-import, wildcard-import, locally-disabled
+from ansible.module_utils.basic import *
+if __name__ == '__main__':
+    main()
```

Comment:
If you haven't submitted this upstream yet, you should! this would be a good candidate for ansible-module-extras.